src/main/java/edu/gatech/cs2340/trydent/GameObject.java

```diff
@@ -69,6 +78,71 @@ public GameObject(String name, Node javaFXObject) {
         this.name = name;
     }
 
+    /**
+     * Plays the given animation, stopping any currently playing animations.
+     *
+     * @param animation
+     */
+    public void playAnimation(Animation animation) {
+        this.animation = animation;
+        this.animationStartTime = Time.getTime();
+        this.animationLoopCounter = 1;
+        animation.setTimeWrap(TimeWrapMode.CLAMP);
+    }
+
+    /**
+     * Plays the given animation, stopping any currently playing animations, and
+     * loops it count number of times.
+     *
+     * @param animation
+     * @param count
+     *            The number of times to play the animation. count = 0 will not
+     *            play anything, count = 1 will play the animation exactly once.
+     *            A negative value for count will cause the animation to loop
+     *            infinitely.
+     */
+    public void loopAnimation(Animation animation, int count) {
+        this.animation = animation;
+        this.animationStartTime = Time.getTime();
+        this.animationLoopCounter = count;
+        animation.setTimeWrap(TimeWrapMode.WRAP);
+    }
+
+    /**
+     * Plays the given animation in a loop forever.
+     *
+     * @param animation
+     */
+    public void loopAnimation(Animation animation) {
+        loopAnimation(animation, -1);
+    }
+
+    /**
+     * Stops the currently playing animation.
+     */
+    public void stopAnimation() {
+        this.animation = null;
+    }
+
+    private void createAnimationBehavior() {
+        new Behavior(this) {
+            @Override
+            public void onUpdate() {
+                GameObject g = getGameObject();
+                if (g.animation == null) {
+                    return;
+                }
+                double time = Time.getTime() - g.animationStartTime;
+                if (g.animationLoopCounter >= 0 && time > g.animationLoopCounter * g.animation.getDuration()) {
+                    g.animation = null;
+                    // TODO: add callbacks saying that the animation ended.
```

Comment:
Can just call Behavior.stop(), or is there something else that needs to happen here?